setup.py

```diff
@@ -57,7 +57,27 @@ def find_flux():
     return os.path.dirname(os.path.dirname(os.path.realpath(path)))
 
 
-flux_root = find_flux()
+def is_info_command():
+    """
+    Check if we're running a command that only needs package metadata
+    and doesn't require Flux to be installed.
+    """
+    info_commands = {
+        "egg_info",
+        "sdist",
+        "--name",
+        "--description",
+        "--help",
+        "--help-commands",
+        "--version",
+    }
+    return any(cmd in sys.argv for cmd in info_commands)
```

Comment:
One more addition to account for `python setup.py` acting like help, then I think we've covered the examples she provided.

```suggestion
    # This is akin to calling --help
    if len(sys.argv) == 1 and sys.argv[0] == "setup.py":
        return True
    return any(cmd in sys.argv for cmd in info_commands)
```